log/logger.go

```diff
@@ -1,12 +1,89 @@
 package log
 
-import cmtlog "github.com/cometbft/cometbft/libs/log"
+import (
+	"os"
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+// Defines commons keys for logging
+const ModuleKey = "module"
+
+// ContextKey is used to store the logger in the context
+var ContextKey struct{}
 
-// Logger is the interface that wraps the basic logging methods.
 type Logger interface {
```

Comment:
can we add a godoc to this inteface, 

if we are breaking the interface here maybe we put together an interface we want instead of adopting this from comet? Mainly making it typed instead of untyped

cc @alexanderbez

Reply:
Yes, for backward compatibility I adopted the main interface for comet, so people just need to change their imports.
However, the structured API can be be used by calling the `Impl()` method, which surfaces the zerolog implementation.
Otherwise, we are very tightly coupled to zerolog API, which almost makes no sense to define our own interface instead of directly using zerolog types.

Reply:
well the idea is that an application can define their own logger if they'd like. We cant know if its zerolog or something else. That is why the interface uses interface{} everywhere. Id prefer to keep this level of modularity for users.

Reply:
Sure, I've changed the `Impl()` to return an interface then (https://github.com/cosmos/cosmos-sdk/pull/15011/commits/e8f244012ae0f41be4b5c876ac03e2df166693d4)
So, we can still type cast the implementation to zerolog.Logger to use the structured API from zerolog, without having the interface too coupled to zerolog.
This way, we really have no dependencies and people are free to define their logger.

Reply:
this would mean that if we use impl then users cant swap it out for another logger that has a different api?

It seems this is the same as before. We can wait for others to chime in but i think we should leave the interface as is for  this pr since its already touching a lot and do another pr that breaks the interface if we want to.

Reply:
No, it means that we can type cast in the SDK and decide when it makes sense to use the structured API for the perf increase than when using the `Fields()` method from zerolog.
We can always fall back to call the default API `Debug(), Info()` so each logger is supported.

Reply:
should we then provide helpers for zerolog? 

Worried that a user or dev can interface cast without checking if the cast is successful. we dont do this in the tests since we know what it will return correctly.